src/py42/services/legalholdapiclient.py

```diff
@@ -0,0 +1,373 @@
+from py42 import settings
+from py42.exceptions import Py42BadRequestError
+from py42.exceptions import Py42Error
+from py42.exceptions import Py42ForbiddenError
+from py42.exceptions import Py42LegalHoldAlreadyActiveError
+from py42.exceptions import Py42LegalHoldAlreadyDeactivatedError
+from py42.exceptions import Py42LegalHoldCriteriaMissingError
+from py42.exceptions import Py42LegalHoldNotFoundOrPermissionDeniedError
+from py42.exceptions import Py42UserAlreadyAddedError
+from py42.services import BaseService
+from py42.services.util import get_all_pages
+
+
+def _active_state_map(active):
+    _map = {True: "ACTIVE", False: "INACTIVE", None: "ALL"}
+    try:
+        return _map[active]
+    except KeyError:
+        raise Py42Error(
+            f"Invalid argument: '{active}'. active must be True, False, or None"
+        )
+
+
+class LegalHoldApiClientService(BaseService):
+    """A service for interacting with Code42 Legal Hold APIs.
+
+    The LegalHoldService provides the ability to manage Code42 Legal Hold Policies and Matters.
+    It can:
+    - Create, view, and list all existing Policies.
+    - Create, view, deactivate, reactivate, and list all existing Matters.
+    - Add/remove Custodians from a Matter.
+    """
+
+    _uri_prefix = "/api/v27"
```

Comment:
is v27 correct here? I know we don't follow the typical versioning, but I feel like I should still ask.

Reply:
Yeah, this isn't our versioning scheme, these endpoints are provided by the microservice (or in this case core) and don't go through the gateway.